### PR TITLE
Add targets-based workflow vignette to dataRetrieval

### PR DESCRIPTION
**Pull request by padilla410**
## Overview

Howdy Laura,

I am opening this PR to start a conversation around remaining steps I need to complete before merging in the targets & dataRetrieval vignette that was posted on WDFN [here](https://waterdata.usgs.gov/blog/wqp-large-pull-targets/).

I added this vignette "as is" based on our conversation in early October.

## Summary of changes
* I added the targets workflow Rmd and the associated images. All of the content is the same as the WDFN post.

## Verification & testing
* I tested this locally by knitting and then opening in the browser to inspect the elements and verify that the expected alt-text was present. Here are some screenshots:

image

image

## Review Request
* Is this what you were expecting? If not, please let me know and I can make the edits.
* It has been a while since I've done package development. Could you point me the latest and greatest docs for running whatever tests you would like me to run?

### CODE REVIEW

**Review by ldecicco-USGS — COMMENTED** [review]
We'll also need to add a line in the "Articles" pkgdown file:
https://github.com/USGS-R/dataRetrieval/blob/main/_pkgdown.yml#L25
(right below line 36)

I'm going to suggest adding all these new files to the ".Rbuildignore" file. The article will still then be included in all the dataRetrieval package documentation, but it won't be sent to CRAN. I try to minimize CRAN submission files to avoid any false CRAN flags. Rest assured, it seems like people prefer our pkgdown documentation to the embedded vignettes anyway.

File: vignettes/wqp_large_pull_targets.Rmd [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+enables the access and retrieval of over 297,000,000 water quality
+records (Read et al. 2017) through web services and an application
+programming interface (API) that can be called programmatically using
+the
+[`dataRetrieval`](https://cran.r-project.org/web/packages/dataRetrieval/vignettes/dataRetrieval.html)
+package in R. Downloading data from the WQP represents a common pattern
+across USGS data teams. 
+
+In this post, we highlight an [example data
+pipeline](https://github.com/USGS-R/ds-pipelines-targets-example-wqp/) to increase the reusability, reproducibility, and efficiency of
```
[the remark this answers is not in the document]
Comment (padilla410) [reply vignettes/wqp_large_pull_targets.Rmd:33]: Y'know, I considered this, but we just aren't ready to commit to a new home yet. What I will do is open an issue in the current repo. If/when we move to GitLab, that issue will come along for the ride and I will close it out by updating this repo.

File: vignettes/wqp_large_pull_targets.Rmd [hunk 2]
```diff
@@ -0,0 +1,266 @@
+---
+title: "Large Data Pulls from Water Quality Portal Using `dataRetrieval` - A Pipeline-Based Approach"
+author: "Lauren Koenig (she/her), Lindsay Platt (she/her), Julie Padilla (she/her)"
+date: "2022-09-27"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
```
[the remark this answers is not in the document]
Comment (padilla410) [reply vignettes/wqp_large_pull_targets.Rmd]: [a74d597](https://github.com/USGS-R/dataRetrieval/pull/640/commits/a74d597b47e2218309482932d748deaaea54d977)

File: vignettes/wqp_large_pull_targets.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (padilla410) [reply vignettes/wqp_large_pull_targets.Rmd:33]: https://github.com/USGS-R/ds-pipelines-targets-example-wqp/issues/98

**Comment by padilla410** [comment]
Howdy Laura,

I either made the suggested edits or addressed you comments. Looks like I am still failing some checks, but looking at the checks, it looks like it's due to testthat failures that are unrelated to this vignette. If that is _not_ the case, please let me know and I will make the required edits.

**Pull request merged**